app/frontend/src/features/communities/discussions/CommentForm.tsx

```diff
@@ -0,0 +1,114 @@
+import { Collapse, Typography } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import MarkdownInput from "components/MarkdownInput";
+import { Error as GrpcError } from "grpc-web";
+import { PostReplyRes } from "pb/threads_pb";
+import { threadKey } from "queryKeys";
+import { MouseEventHandler, useState } from "react";
+import { useForm } from "react-hook-form";
+import { useMutation, useQueryClient } from "react-query";
+import { service } from "service";
+import makeStyles from "utils/makeStyles";
+
+import { CLOSE, COMMENT, WRITE_COMMENT_A11Y_LABEL } from "../constants";
+
+const useStyles = makeStyles((theme) => ({
+  commentForm: {
+    display: "flex",
+    flexDirection: "column",
+    "& > :not(:last-child)": {
+      marginBlockEnd: theme.spacing(1),
+    },
+    gridArea: "commentForm",
+  },
+  buttonsContainer: {
+    display: "flex",
+    justifyContent: "flex-end",
+    "& > * + *": {
+      marginInlineStart: theme.spacing(2),
+    },
+  },
+}));
+
+interface CommentFormProps {
+  shown?: boolean;
+  testId: string;
+  threadId: number;
+}
+
+interface CommentData {
+  content: string;
+}
+
+export default function CommentForm({
+  shown = false,
+  testId,
+  threadId,
+}: CommentFormProps) {
+  const classes = useStyles();
+  const [showCommentForm, setShowCommentForm] = useState(shown);
+  const { control, handleSubmit, reset: resetForm } = useForm<CommentData>({
+    mode: "onBlur",
+  });
+  const queryClient = useQueryClient();
+  const {
+    error,
+    isLoading,
+    mutate: postComment,
+    reset: resetMutation,
+  } = useMutation<PostReplyRes.AsObject, GrpcError, CommentData>(
+    ({ content }) => service.threads.postReply(threadId, content),
+    {
+      onSuccess() {
+        queryClient.invalidateQueries(threadKey(threadId));
```

Comment:
We went with showing a refetch loading state instead, since optimistically updating the data in an infinite query is much more of a pain than a normal query. There is one in `Comment`, but I may have to adjust the position of it to be at the bottom if we are going with having the comment input near the last comment.